### ISSUE TITLE
[Feature]: Change implementation of containers types to simplify API

### ISSUE THREAD

**Issue by ehennestad** [NeurodataWithoutBorders/matnwb#625]
### What would you like to see added to MatNWB?

Many NWB types are containers for other types, with two examples being `ProcessingModule` and `Fluorescence`.

The syntax for setting/getting data for these containers is cumbersome:
```
ophys_module = types.core.ProcessingModule('description', 'contains optical physiology data')

ophys_module.nwbdatainterface.set('ImageSegmentation', img_seg);
```
and 

```
fluorescence = types.core.Fluorescence();

fluorescence.roiresponseseries.set('RoiResponseSeries', roi_response_series);
ophys_module.nwbdatainterface.set('Fluorescence', fluorescence);
```

A simpler way of doing this would be:
```
ophys_module = types.core.ProcessingModule('description', 'contains optical physiology data')

ophys_module('ImageSegmentation') = img_seg;
```
and
```
fluorescence = types.core.Fluorescence();

fluorescence('RoiResponseSeries') = roi_response_series;
ophys_module('Fluorescence') = fluorescence;
```

A similar suggestion was raised in pynwb ([issue #1993](https://github.com/NeurodataWithoutBorders/pynwb/issues/1993)) and it would be great to mirror this on the matnwb side.

### What solution would you like?

Redefine container classes to have `dictionary`/`Set` like behavior, while also restricting the types it is allowed to hold

### Do you have any interest in helping implement the feature?

Yes.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/neurodatawithoutborders/matnwb/blob/master/.github/CODE_OF_CONDUCT.md)
- [X] Have you ensured this change was not already [requested](https://github.com/NeurodataWithoutBorders/matnwb/issues)?

**Comment by bendichter** [comment]
@ehennestad this sounds like a nice change, as long as we maintain backwards compatibility and have good tests for both the old and the new mode.

**Comment by ehennestad** [comment]
> But note that this only works when a container has a single Set object, so we'll need to check for that and revert to only supporting the old style if there are multiple.

Can you explain what you mean? Is this referring to the ProcessingModule where there are sets for both nwbdataInterface and dynamic tables?

**Comment by bendichter** [comment]
Yeah, I guess I was thinking `pm.nwbdatainterface.get('this')` would collide with `pm.dynamictables.get('this')`, but now that I think about it you can't store two different objects within the same Group, so you would never run into that problem.
